### PR TITLE
Add static members to the extended completion

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://github.com/Microsoft/visualfsharp/issues/3948

image

### CODE REVIEW

File: src/fsharp/vs/ServiceAssemblyContent.fs [hunk 1]
```diff
@@ -254,7 +255,7 @@ module AssemblyContentProvider =
                     | None -> ()
 
                     let thisRequiresQualifierAccess =
-                        if entity.IsFSharp && Symbol.hasAttribute<RequireQualifiedAccessAttribute> entity.Attributes then 
+                        if (entity.IsFSharp && Symbol.hasAttribute<RequireQualifiedAccessAttribute> entity.Attributes) || entity.IsClass then
```
Comment (auduchinok) [inline src/fsharp/vs/ServiceAssemblyContent.fs]: We should count value types and simple types that may have static members as well.

Comment (vasily-kirichenko) [reply src/fsharp/vs/ServiceAssemblyContent.fs]: I know :) The problem that static members should always be prefixed with the type (any type) name, but module values - if the module is RQA.

**Comment by auduchinok** [comment]
@vasily-kirichenko Do we need all members? I remember ReSharper adding such completion for properties/fields only and am OK with it this way. I don't think I'd call a method this way from completion.

**Comment by vasily-kirichenko** [comment]
Fixed.

@auduchinok why not?

image

**Comment by auduchinok** [comment]
@vasily-kirichenko I'm a bit afraid of how many members there may be when lots of assemblies are referenced. Probably a good compromise would be to make it optional.

**Comment by cartermp** [comment]
We'll do some acceptance testing on projects with a bunch of assemblies before pulling this.  I imagine perf would be proportional to how it is today (unless there are tons of solutions out there which use crazy amounts of static members compared with instance members, which I doubt is very common).

**Comment by vasily-kirichenko** [comment]
It's ready.

**Comment by saul** [comment]
Why are we showing static members? Surely these should only show after you’ve dotted into the type that has the static members?

**Comment by smoothdeveloper** [comment]
@saul, it is the same idea of having the namespaces open when you put a type name from a namespace which is not in scope. 

Since F# has few places where static member needs the explicit prefix (in instance members of a same class) and in more general cases (like looking for an extension method which is not in scope), I think that feature is useful.

**Comment by smoothdeveloper** [comment]
@vasily-kirichenko thanks for that feature! is there a special handling for the extension methods already or can we consider adding it?

Here is what Resharper does in C# code, here it shows extension methods that are not in scope.

image

I tried the same in recently updated VS2017 and it doesn't have that, but maybe I need to install a nighlty?

**Comment by vasily-kirichenko** [comment]
@smoothdeveloper extention members are not supported. To do it we need to collect all of them in a separate list in assembly content cache and mix into after-dot completion.

**Comment by vasily-kirichenko** [comment]
@saul because ReSharper does this and it would be great to be on par with C# completion in Rider. Personally I cannot say if this feature is useful or not, will see,

**Comment by vasily-kirichenko** [comment]
Ooookey, it's ready.

**Pull request merged**